### PR TITLE
feat: support bootstrapping AWS clusters

### PR DESCRIPTION
**Pull request by Unix4ever**
New commands:

```bash
$ capi bootstrap capi # renamed from capi bootstrap
$ capi bootstrap cluster --provider aws --name cluster --from template/template.yaml
```

Using alternate `AWS` cluster template by default which is almost the
same as our template in `templates` repo, but as all AWS specific stuff
prefixed by `AWS_` and is using common CAPI names for workers and
control planes count.

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

### CODE REVIEW

File: pkg/capi/cluster.go [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+		schema.GroupVersionKind{
+			Version: version,
+			Group:   "cluster.x-k8s.io",
+			Kind:    "Machine",
+		},
+	)
+
+	talosConfig.SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Version: version,
```
Comment (Unix4ever) [inline pkg/capi/cluster.go:106]: has own version

File: pkg/capi/cluster.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+	controlPlaneRef, err := getNamespacedName(cluster, "spec", "controlPlaneRef")
+	if err != nil {
+		return nil, err
+	}
+
+	if err = clusterAPI.runtimeClient.Get(ctx, controlPlaneRef, &controlPlane); err != nil {
+		return nil, err
+	}
+
+	if val, err = getField(controlPlane, "status", "selector"); err != nil {
```
Comment (Unix4ever) [inline pkg/capi/cluster.go:132]: Check if unstructured has recursive get.

File: pkg/capi/cluster.go [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+	}
+
+	var (
+		clientConfig      *clientconfig.Config
+		ok                bool
+		talosConfigString string
+	)
+
+	val, err = getField(talosConfig, "status", "talosConfig")
+	if err != nil {
```
Comment (Unix4ever) [inline pkg/capi/cluster.go:165]: That's machine config, no need to pull it now

File: pkg/capi/cluster.go [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+	}
+
+	nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type != v1.NodeExternalIP {
```
Comment (Unix4ever) [inline pkg/capi/cluster.go:233]: Add all internal IPs to `cp/worker` endpoints. 
ExternalIP to controlPlanes for talosconfig.

File: pkg/capi/infrastructure/aws/aws.yaml [hunk 5]
```diff
@@ -0,0 +1,186 @@
+## Cluster configs
+apiVersion: cluster.x-k8s.io/v1alpha3
```
Comment (rsmitty) [inline pkg/capi/infrastructure/aws/aws.yaml:2]: We'll either want to roll these changes back out to the `cluster-api-templates` repo or get rid of that repo and use this one as a source of truth. If we do the latter, we'll need to get the multi-az and autoscaling templates worked in here too. Just something to note for the future.

**Review by rsmitty — APPROVED** [review]

**Review by smira — APPROVED** [review]
:cool:

**Pull request merged**